datadog_lambda/tracing.py

```diff
@@ -90,6 +89,27 @@ def _context_obj_to_headers(obj):
     }
 
 
+def create_dd_dummy_metadata_subsegment(
+    dd_context, trace_context_source, subsegment_metadata_key
+):
+    """
+    Create a Datadog subsegment to pass the Datadog trace context or Lambda function
+    tags into its metadata field, so the X-Ray trace can be converted to a Datadog
+    trace in the Datadog backend with the correct context.
+    """
+    if not dd_context or (
+        trace_context_source and trace_context_source != TraceContextSource.EVENT
```

Comment:
I not sure this check on `trace_context_source` belongs to this general helper function, e.g., is it needed for both use cases? 

I think you should do the check outside before calling this function.